<img style="float: left; margin: 30px 15px 15px 15px;" src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFzQj91sOlkeDFkg5HDbjtR4QJYmLXkfMNig&usqp=CAU" width="400" height="500" /> 
    
    
### <font color='navy'> Simulación matemática. 

**Nombres:** Jorge Emilio Andrade Michel, Ramón Rojo Villalobos.

**Fecha:** 02 de octubre del 2023.

**Expediente** : 739973, 744371.
    
**Profesor:** Oscar David Jaramillo Zuluaga.

**Link de GitHub:** https://github.com/RamonRojoVi/Tarea-5
# Tarea 5: Integración Montecarlo 1

## TAREA 5
1. Realizar las siguientes integrales. Poner los resultados en una tabla cuyas filas correspondan a la cantidad de términos utilizados en la aproximación (usar 10, 100, 1000, 10000 y 100000 términos) y cuyas columnas correspondan a las funciones.
    + $\int_{1}^{5} \log_{10}(2x)+8^x\text{d}x$.
    + $\int_1^5 \frac{2^x}{x^4} \text{d}x$


#### Codigo E1

In [2]:
import numpy as np
import pandas as pd

# Definición de la función montecarlo1
def montecarlo1(f, a, b, N):
    ui = np.random.uniform(a, b, N)
    integral = ((b-a)/N)*np.sum(f(ui))
    return integral

# Definición de las funciones que vamos a integrar
def f4(x):
    return np.log10(2*x) + 8**x

def f5(x):
    return (2*x) / (x*4)

# Definimos los limites de integracion
a, b = 1, 5

# Definimos cantidades de terminos
N = np.logspace(1, 5, 5).astype(int)

# Crear el DataFrame de pandas (tabla) que contendrá la información
tabla3 = pd.DataFrame(index=N, columns=['Aprox f4', 'Aprox f5'])

# Agregar nombre al índice para que aprezca en la tabla
tabla3.index.name = 'Cantidad de términos'

# Rellenar las celdas de la aproximación de cada función
for n in N:
    aprox_f4 = montecarlo1(f4, a, b, n)
    aprox_f5 = montecarlo1(f5, a, b, n)
    
    tabla3.loc[n, 'Aprox f4'] = aprox_f4
    tabla3.loc[n, 'Aprox f5'] = aprox_f5
    
tabla3

Aprox f4 Aprox f5
Cantidad de términos                       
10                     4762.616066      2.0
100                    16827.52595      2.0
1000                  16146.823994      2.0
10000                 16121.820848      2.0
100000                15734.670965      2.0

##### Conclusión de resultados:


#### Código E2

In [4]:
import numpy as np
import pandas as pd

# Definición de las funciones a integrar
def f1(x):
    return np.log10(2*x) + 8**x

def f2(x):
    return 2**x / x**4

# Definición de la función de Montecarlo tipo 1
def montecarlo1(f, a, b, N):
    ui = np.random.uniform(a, b, N)
    integral = ((b-a)/N)*np.sum(f(ui))
    return integral

# Límites de integración
a, b = 1, 5

# Cantidades de términos para la aproximación
N = np.logspace(1, 5, 5).astype(int)

# Crear una tabla (DataFrame) para almacenar las aproximaciones
tabla = pd.DataFrame(index=N, columns=['Aproximación f1', 'Aproximación f2'])
tabla.index.name = 'Cantidad de términos'

# Rellenar la tabla con las aproximaciones
for n in N:
    tabla.loc[n, 'Aproximación f1'] = montecarlo1(f1, a, b, n)
    tabla.loc[n, 'Aproximación f2'] = montecarlo1(f2, a, b, n)

tabla


Aproximación f1 Aproximación f2
Cantidad de términos                                
10                      19299.204831        0.931824
100                     14275.654903        1.122116
1000                    15374.378497        1.009295
10000                   15668.658392        0.993039
100000                  15917.770681        1.003553

##### Conclusión de resultados:
Los resultados obtenidos de la aproximación de las integrales presentan comportamientos notablemente distintos para las dos funciones evaluadas, $f2$, utilizando diferentes cantidades de términos en el método de Monte Carlo.

Para la función $f1$, observamos una disminución inicial en la aproximación de la integral al aumentar el número de términos de 10 (19299.20) a 100 (14275.65), seguido de un incremento gradual al aumentar aún más el número de términos a 1000 (15374.37), 10000 (15668.65) y 100000 (15917.77). Esta tendencia podría indicar que la aproximación está convergiendo hacia un valor específico a medida que se utilizan más términos, pero aún se requiere exploración adicional para determinar si la aproximación continúa estabilizándose o cambia al utilizar un número aún mayor de términos.

En contraste, para la función $f2$, los valores de la aproximación fluctúan alrededor de 1, con un valor máximo de 1.122116 con 100 términos y un mínimo de 0.993039 con 10000 términos. Este comportamiento sugiere que la integral podría estar convergiendo a un valor cercano a 1, con pequeñas fluctuaciones en la aproximación dependiendo del número de términos utilizados.

En resumen, estos resultados enfatizan la importancia de elegir adecuadamente el número de términos al utilizar el método de Monte Carlo para aproximar integrales. Mientras que la función $f2$ muestra signos de convergencia rápida, la función $f1$ presenta un comportamiento más complejo que podría beneficiarse de una exploración adicional para determinar la cantidad óptima de términos que garantiza una estimación precisa y confiable de la integral.